File: Solutions/estimate_pi.go
package main

import (
	"fmt"
	"math"
	"math/rand"
)

func main() {
	var estimatedPi float64 = estimatePi(10000000)
	fmt.Println(estimatedPi)
}

func estimatePi(n int) float64 {
	numOfPointsCircle := 0
	numOfPointsTotal := 0

	for i := 0; i < n; i++ {
		var x float64 = rand.Float64()
		var y float64 = rand.Float64()

		distance := math.Sqrt((x * x) + (y * y))

		if distance <= 1 {
			numOfPointsCircle++
		}
		numOfPointsTotal++
	}

	var estimate float64 = (4 * (float64(numOfPointsCircle) / float64(numOfPointsTotal)))

	return estimate
}

File: Solutions/estimate_pi.c
// Estimate pi, given that you have random (0,1)

#include <stdio.h>
#include <stdlib.h>
#include <math.h>

float estimatePi(int n){
    int numOfPoints_Circle = 0;
    int numOfPoints_Total = 0;

    for (int i = 0; i < n; i++){
        float x = (float)rand()/RAND_MAX;
        float y = (float)rand()/RAND_MAX;

        float distance = sqrt((x * x) + (y * y));

        if (distance <= 1){
            numOfPoints_Circle++;
        }
        numOfPoints_Total++;
    }

    float estimate = (4 * ((float)numOfPoints_Circle / (float)numOfPoints_Total));
    return estimate;
}

int main(){
    float estimatedPi = estimatePi(10000000);
    printf("%f", estimatedPi);
}
